src/wagtailtrans/models.py

```diff
@@ -87,22 +86,29 @@ def _language_default():
 
 @python_2_unicode_compatible
 class TranslatablePage(Page):
+    translatable_page_ptr = models.OneToOneField(
+        Page, parent_link=True, related_name='+', on_delete=models.CASCADE)
     canonical_page = models.ForeignKey(
         'self', related_name='translations', blank=True,
         null=True, on_delete=models.SET_NULL)
     language = models.ForeignKey(
         Language, related_name='pages', on_delete=models.PROTECT,
         default=_language_default)
 
-    translation_panels = [
-        MultiFieldPanel([
-            FieldPanel('language'),
-            PageChooserPanel('canonical_page'),
-        ])
+    settings_panels = Page.settings_panels + [
+        MultiFieldPanel(
+            heading=_("Translations"),
+            children=[
+                FieldPanel('language'),
+                PageChooserPanel('canonical_page'),
+            ]
+        )
     ]
 
     base_form_class = AdminTranslatablePageForm
 
+    is_creatable = False
```

Comment:
No we don't want it to be abstract since we're doing some queries on the model..